quartz/components/scripts/decrypt.inline.ts

```diff
@@ -0,0 +1,168 @@
+import { base64 } from "rfc4648"
+
+// @ts-ignore:next-line
+function find<T>(selector: string): T {
+  const element = document.querySelector(selector) as T
+  if (element) return element
+}
+
+let salt: Uint8Array, iv: Uint8Array, ciphertext: Uint8Array, iterations: number
+
+async function decryptHTML() {
+  const pl = find<HTMLPreElement>("pre[data-i]")
+  if (!pl) {
+    return
+  }
+
+  const form = find<HTMLFormElement>("form")
+  if (!form) {
+    return
+  }
+
+  const pwd = find<HTMLInputElement>(".pwd")
+  if (!pwd) {
+    return
+  }
+
+  form.addEventListener("submit", async (event) => {
+    event.preventDefault()
+    await decrypt()
+  })
+  if (!subtle) {
+    error("modern")
+    pwd.disabled = true
+  }
+
+  show(find<HTMLDivElement>("#lock"))
+  if (!pl.innerHTML) {
+    pwd.disabled = true
+    error("empty")
+    return
+  }
+  iterations = Number(pl.dataset.i)
+  const bytes = base64.parse(pl.innerHTML)
+  salt = bytes.slice(0, 32)
+  iv = bytes.slice(32, 32 + 16)
+  ciphertext = bytes.slice(32 + 16)
+
+  if (location.hash) {
+    const parts = location.href.split("#")
+    pwd.value = parts[1]
+    history.replaceState(null, "", parts[0])
+  }
+
+  if (sessionStorage[document.body.dataset.slug!] || pwd.value) {
+    await decrypt()
+  } else {
+    hide(find<HTMLDivElement>("#load"))
+    show(form)
+    pwd.focus()
+  }
+}
+
+document.addEventListener("DOMContentLoaded", decryptHTML)
```

Comment:
why do we need to add event for both nav and domcontentloaded here?

afaict this is required in quartz

Reply:
Just for testing, removed